torchmultimodal/models/omnivore.py

```diff
@@ -104,11 +115,97 @@ def _omnivore_swin_t_encoder() -> SwinTransformer3d:
     return encoder
 
 
-# TODO: add pretrained weight capability
-def omnivore_swin_t(encoder_only=False) -> nn.Module:
+def _omnivore_swin_s_encoder() -> SwinTransformer3d:
+    encoder = SwinTransformer3d(
+        patch_size=[2, 4, 4],
+        embed_dim=96,
+        depths=[2, 2, 18, 2],
+        num_heads=[3, 6, 12, 24],
+        window_size=[8, 7, 7],
+        stochastic_depth_prob=0.3,
+        norm_layer=nn.LayerNorm,
+        patch_embed=PatchEmbedOmnivore,
+        num_classes=None,
+    )
+    return encoder
+
+
+def _omnivore_swin_b_encoder() -> SwinTransformer3d:
+    encoder = SwinTransformer3d(
+        patch_size=[2, 4, 4],
+        embed_dim=128,
+        depths=[2, 2, 18, 2],
+        num_heads=[4, 8, 16, 32],
+        window_size=[16, 7, 7],
+        stochastic_depth_prob=0.3,
+        norm_layer=nn.LayerNorm,
+        patch_embed=PatchEmbedOmnivore,
+        num_classes=None,
+    )
+    return encoder
+
+
+def omnivore_swin_t(
+    encoder_only: bool = False, pretrained: bool = False, progress: bool = True
+) -> nn.Module:
+    """
+    Builder function to get omnivore model with swin_t variant encoder
+    Args:
+        encoder_only (bool): If true then the builder will only return encoder without head (default: False)
+        pretrained (bool): If true then the it will load pretrained weight,
+            otherwise it will have random weight (default: False)
+        progress (bool): If true then there will be a progress bar for downloading weight (default: True)
+    """
     encoder = _omnivore_swin_t_encoder()
+    heads = _multimodal_head(input_dim=encoder.num_features)
+    model = OmnivoreArchitecture(encoder, heads)
+    if pretrained:
+        model.load_model(_OMNIVORE_PRETRAINED_URLS["swin_t"])
     if encoder_only:
-        return encoder
+        return model.encoder
```

Comment:
instead of this flag, people can just call _omnivore_swin_t_encoder() right if u allow ckpt loading in that function ?

Reply:
i think same ckpt should work if u set strict to false. try?

Reply:
I have tried using `strict=False` and it still failed because the checkpoint key has `encoder.` prefix compared to what encoder actually expected.
For this case, I am thinking of storing `encoder` and `head` weight separately, and let the encoder and head load separately. But for this, I am thinking of creating utils for `load_model_from_url` since both model and encoder would not have `Mixin` (since the head dont have its own class, and encoder will be upstreamed and it shouldn't inherit PretrainedMixin).
What do you think about this approach @ankitade ?

Reply:
I see the issue. its fine to leave as is for this PR. 
util is a good idea but how will u generalize it? would be good to put up a draft pr/notebook